yaas/host.py

```diff
@@ -1,4 +1,87 @@
 # coding: utf-8
 
+from __future__ import absolute_import
+from __future__ import print_function
+
+import inspect
+import pprint
+import requests
+
+from . import config
+
+
 def command(parser, args):
-    pass
+    """ Interact with registered agents. """
+    subparsers = parser.add_subparsers(dest='subcommand')
+    subcommands = {
+        'list': _list,
+        'show': _show
+        }
+    for name, fn in subcommands.items():
+        subparsers.add_parser(name, help=inspect.getdoc(fn))
+    subargs, extra = parser.parse_known_args(args)
+    subcommands[subargs.subcommand](
+        subparsers.choices[subargs.subcommand],
+        extra)
+
+
+def _list(parser, args):
+    """ List all registered agents. """
+    response = requests.get(
+        config.href('/api/v1/hosts'),
+        **config.requests_opts())
+    if config.args.raw:
```

Comment:
You should deal with failure cases before this, even though there shouldn't be any. See comment below